base/multidimensional.jl

```diff
@@ -940,3 +940,178 @@ If `dim` is specified, returns unique regions of the array `itr` along `dim`.
         @nref $N A d->d == dim ? sort!(uniquerows) : (1:size(A, d))
     end
 end
+
+"""
+    groupslices(A, dim)
+
+Returns a vector of integers where each integer element of the returned vector
+is a group number corresponding to the unique slices along dimension `dim` as
+returned from `unique(A, dim)`, where `A` can be a multidimensional array.
+
+Example usage:
+
+If `C = unique(A, dim)`, `ic = groupslices(A, dim)`, and
+rank(A) == rank(C) == 3, then:
```

Comment:
should have backticks, and I think you mean `ndims` here not linear algebra rank